static-analysis/teammates-pmd.xml

```diff
@@ -27,6 +27,8 @@
         <exclude name="SimpleDateFormatNeedsLocale"/>
         <exclude name="UseLocaleWithCaseConversions"/>
         <exclude name="UseVarargs"/>
+        <exclude name="DataClass"/>
+        <exclude name="OptimizableToArrayCall"/>
```

Comment:
This is a good rule and should not be suppressed. PMD did say that their stance on this rule changes as they go to PMD 6, but that would mean the suppression should be targeted to the PMD 5 ruleset.

Reply:
Actually I did mention this in https://github.com/TEAMMATES/teammates/issues/8512#issue-299563728, and suggested that the current implementations can be modified at a later issue (obviously I did not know about the eclipse issue back then and neither thought of having a frozen config file for the time being so assumed this should be a separate issue where we can modify the current implementations and then re-enforce this rule.)